### PR TITLE
Fix taiko scroller not correctly adhering to gameplay time movement

### LINKED ISSUE #8995: taiko scroller moves even if paused
**Describe the bug:** title

**Screenshots or videos showing encountered issue:** https://drive.google.com/open?id=1y6CWsv5rwYsQN3d4_F-mWOmp6QF_tlAj

**osu!lazer version:** aa91288

**Logs:**
[logs.zip](https://github.com/ppy/osu/files/4608237/logs.zip)

### PR DESCRIPTION
**Pull request by peppy**
- Closes https://github.com/ppy/osu/issues/8995
- Adds forgotten resources bump to make the scroller actually appear for default skin

More of a large change than I hoped for, but I think this method reads better. Maybe.

### CODE REVIEW

File: osu.Game.Rulesets.Taiko/Skinning/LegacyTaikoScroller.cs [hunk 1]
```diff
@@ -56,31 +56,37 @@ protected override void Update()
         {
             base.Update();
 
-            while (true)
-            {
-                float? additiveX = null;
+            bool wideEnough() =>
+                InternalChildren.Any()
+                && InternalChildren.First().ScreenSpaceDrawQuad.Width * InternalChildren.Count >= ScreenSpaceDrawQuad.Width * 2;
```
[the remark this answers is not in the document]
Comment (bdach) [reply osu.Game.Rulesets.Taiko/Skinning/LegacyTaikoScroller.cs:61]: Once texture wrapping makes it in framework-side we might potentially look at removing this type of logic altogether in favour of just animating texture coords instead? Not sure how viable that is.

[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game.Rulesets.Taiko/Skinning/LegacyTaikoScroller.cs:61]: Yeah that could work.

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Rulesets.Taiko/Skinning/LegacyTaikoScroller.cs:61]: There's zero overhead of off-screen sprites? I did consider this.

Using texture wrapping may work or may be too hacky. We'll see when it's ready.

**Comment by smoogipoo** [comment]
image

This is creating hundreds of sprites.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
As mentioned

**Comment by peppy** [comment]
Can you provide a case that creates hundreds? That sounds insanely wrong. It's only supposed to generate 2x screen width.

**Comment by smoogipoo** [comment]
Any beatmap. Here's one https://osu.ppy.sh/beatmapsets/587439#taiko/1244015

**Comment by peppy** [comment]
Ah, beatmap specific skins?

**Comment by smoogipoo** [comment]
Classic skin

**Comment by peppy** [comment]
Again, that makes zero sense. I've heavily tested and cannot reproduce so please provide more details.

**Comment by peppy** [comment]
Nevermind, looks to be a regression with the seemingly safe last commit.

**Comment by peppy** [comment]
I've fixed the issue by converting from a `while` to `if`. It should not be noticeable under any sane scenario (would see the scroller incorrectly for maybe one frame during initial fade in on a super super ultra wide). Because this is a legacy component and may change with taiko playfield sizing changes (to allow it work work with `while`) or with texture wrapping (for a saner implementation) I see this as the most efficient fix for now.

**Review by smoogipoo — APPROVED** [review]
Seems fine for now, let's see how this goes.

**Pull request merged**